### PR TITLE
Updated docstring with relevant error raise and examples

### PR DESCRIPTION
**Pull request by John-Aradan**
- Added a Type error raise in case of input is not an array-like object.
- Added thorough examples for the working of the function

### CODE REVIEW

**Comment by fspyridakos** [comment]
Hey, thanks for your contribution, could you also update the docstrings in the array, container and backend instances of this function?

**Comment by John-Aradan** [comment]
@fspyridakos 
#12544 

Updates made as requested:

- 
Added docstring for unique_value: array instance
- Added docstring for unique_value: container instance
- Added docstring for unique_value: numpy frontend instance
- Added docstring for unique_value: torch frontend instance
- Added docstring for unique_value: tensorflow frontend instance
- Added docstring for unique_value: jax frontend instance

**Comment by John-Aradan** [comment]
@ivy-seed  Please assign a reviewer for my pull request.

**Review by fspyridakos — CHANGES_REQUESTED** [review]
Looks really good. I must apologize though for not being clear before, the backend variations don't actually need docstrings, they just need the typehints to be correct, which they seem like they already are, could you please remove them? Apologies again

File: ivy/array/set.py [hunk 1]
```diff
@@ -44,6 +44,65 @@ def unique_counts(self: ivy.Array) -> Tuple[ivy.Array, ivy.Array]:
...
+        ------
+        TypeError
+            If `x` is not an instance of `ivy.Array` or `ivy.NativeArray`.
+
+        Examples
+        --------
+        >>> import ivy
+        >>> x = ivy.array([1, 2, 2, 3, 4, 4, 4])
+        >>> y = ivy.unique_values(x)
+        >>> print(y)
```
[the remark this answers is not in the document]
Comment (John-Aradan) [reply ivy/array/set.py]: Its fine 😊 I will get into it right away

[the remark this answers is not in the document]
Comment (John-Aradan) [reply ivy/array/set.py]: @fspyridakos  I have made the necessary changes as asked for. Requesting for a re-review.

**Comment by John-Aradan** [comment]
@fspyridakos I have made the necessary changes as asked for. Requesting for a re-review.

**Comment by fspyridakos** [comment]
Looks good, there seems to be some conflicting changes. Please resolve them and then I'll push 😊

**Comment by John-Aradan** [comment]
@fspyridakos I have resolved the merge conflict. Please review 😊

**Comment by fspyridakos** [comment]
LGTM! Thanks for your contribution!

**Comment by John-Aradan** [comment]
Thankyou ☺️ But, in the subtask set.py unique_values is still not marked as complete

**Pull request merged**